mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -267,7 +272,7 @@ impl TakerSavedSwap {
                 | TakerSwapEvent::MakerPaymentValidateFailed(_)
                 | TakerSwapEvent::TakerPaymentRefunded(_)
                 | TakerSwapEvent::TakerPaymentRefundedByWatcher(_)
-                | TakerSwapEvent::MakerPaymentSpent(_)
+                | TakerSwapEvent::MakerPaymentSpendConfirmed
```

Comment:
I guess version field would be useful in TakerSavedSwap and MakerSavedSwap to simplify such logic

Reply:
> I guess version field would be useful in TakerSavedSwap and MakerSavedSwap to simplify such logic

If I understand this correctly, you are suggesting a new version field for saved swaps not to be confused for swap version for p2p messages. right?

Reply:
Yes, I added a version field in PR#2112, it's a protocol version (for a node to understand what features a remote peer supports)
Here we changed the swap format, I think it's good to have a version field in it, to increment it whenever we change its structure, so when we read old swaps we may understand what features they supported and process accordingly

Reply:
@dimxy @shamardy added version to TakerSavedSwap and MakerSavedSwap https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/commits/f93fcde3244513442856bbfc210f78853e7f1f4a

Reply:
> added version to TakerSavedSwap and MakerSavedSwap

https://github.com/KomodoPlatform/komodo-defi-framework/blob/5ce137fb978e48d5d95332dd75ec3da7d41face9/mm2src/mm2_main/src/lp_swap/recreate_swap_data.rs#L97
https://github.com/KomodoPlatform/komodo-defi-framework/blob/5ce137fb978e48d5d95332dd75ec3da7d41face9/mm2src/mm2_main/src/lp_swap/recreate_swap_data.rs#L301

We can't assume the other side's version in recreate swap data as it can be a different one. I am actually leaning towards removing version all together and do this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199#discussion_r1776229321 suggestion if you approve @dimxy . We shouldn't have more changes to legacy swap files in the future.

Reply:
P.S. the alternative if we kept version field is to provide it here https://github.com/KomodoPlatform/komodo-defi-framework/blob/5ce137fb978e48d5d95332dd75ec3da7d41face9/mm2src/mm2_main/src/lp_swap/recreate_swap_data.rs#L56-L59 which adds more complexitiy to recreating swap data as maker/taker will have to ask taker/maker for thier version which they might not know.

Reply:
> We can't assume the other side's version in recreate swap data as it can be a different one.

Oh, I thought maker and taker should have same version... could you clarify please, when taker (as an example) recreates swap data from maker, maker is actually their(user) coin just for maker operations, right?

> I am actually leaning towards removing version all together and do this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199#discussion_r1776229321 suggestion

Then I agree that this solution is more solid

Reply:
So answering to this comment https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199#discussion_r1787259679

No problem to do a revert, as before adding TakerSavedSwap and MakerSavedSwap versions, I covered `TakerSwapEvent::MakerPaymentSpentByWatcher` in the first approach https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/commits/51c58cbe367dcbbaae28fd48459342d29c5eb90f (so it was ready for review)

Reply:
@shamardy @dimxy please re review is_coverable function
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199/files#diff-3b58e25a3c557aa8a502011591e9a7d56441fd147c2ab072e108902a06ef3076R267

Reply:
> Oh, I thought maker and taker should have same version... could you clarify please, when taker (as an example) recreates swap data from maker, maker is actually their(user) coin just for maker operations, right? Or it works only in State machine structure which we use in v2 protocol?

From docs https://komodoplatform.com/en/docs/komodo-defi-framework/api/v20/recreate_swap_data/

> The recreate_swap_data can assist in the event of local stored swap data being lost due to storage errors related to low disk space or hardware failure, and if required, aid with the refunding of failed swaps.
> | Structure | Type   | Description                                                                                       |
> |-----------|--------|---------------------------------------------------------------------------------------------------|
> | swap      | object | Swap data from other side of trade. For example to recreate a Maker's swap data, the input would be the corresponding Taker's swap data |

Reply:
> To source the opposite side of the trade, please [contact the Komodo Support team on Discord](https://komodoplatform.com/discord). You will need to provide details about the trade you are trying to recover, such as the coins and amounts being traded, the approximate time of the trade, any known transaction IDs involved in the trade, and if available the UUID of the trade.

Yeah, we cant use taker/maker version in recreate maker/taker swap

Reply:
What I meant by adding a version field is to give a hint to the code about how to process saved swaps, like when a node has many swaps stored from the beginning of mm2 history (while new fields and events have been added) - for convenience in future. If we believe we won't be extending legacy swaps anymore then we do not need version field.
(I think it is a good general practice to have a version wherever some structure may change. You may or may not use it).

Regarding recreation swaps from the other side of the trade I think maybe we should not support recreating from older swaps.

Reply:
> Regarding recreation swaps from the other side of the trade I think maybe we should not support recreating from older swaps.

Sometimes @cipig is using the latest dev and the users will still be using the older version in GUIs and we can't say that we don't support this.

Reply:
> If we believe we won't be extending legacy swaps anymore then we do not need version field.

I think we will not extend it more, but nobody can see the future 🤷